docs/managers.rst

```diff
@@ -28,6 +28,44 @@ so you don't have to.
     >>> conn.num_requests
     3
 
+A :class:`.PoolManager` will create a new :doc:`ConnectionPool <pools>`
+when no :doc:`ConnectionPools <pools>` exist with a matching pool key.
+The pool key is derived using the requested URL and the current values
+of the ``connection_pool_kw`` instance variable on :class:`.PoolManager`.
+
+The keys in ``connection_pool_kw`` used when deriving the key are
+configurable. For example, by default the ``source_address`` key is not
+considered.
```

Comment:
I suppose I didn't do a lot of creative thinking and took https://github.com/shazow/urllib3/pull/830#issuecomment-205453250 a little too literally. It makes sense to me to be a part of the key, so I'm fine adding it as well.